File: src/Ruby/task_3/numbers.rb
def sum_and_product_of_digits(number)
    sum = 0
    product = 1
  
    while number > 0
      digit = number % 10
      sum += digit
      product *= digit
      number /= 10
    end
  
    [sum, product]
  end
  
  input = gets.chomp.split
  
  count = input.shift.to_i
  output = ""
  
  (1..count).each do |index|
    num = input[index - 1].to_i
    sum, product = sum_and_product_of_digits(num)
  
    output += "#{index} " if sum < product
  end
  
  puts output.strip
  

File: src/Java/task_3/numbers.java
import java.util.Scanner;

public class numbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println();
        int count = scanner.nextInt();

        for (int i = 1; i <= count; i++) {
            int num = scanner.nextInt();
            if (isSumLessThanProduct(num)) {
                System.out.print(i + " ");
            }
        }
        scanner.close();
        System.out.println();
    }

    private static boolean isSumLessThanProduct(int num) {
        int sum = 0, product = 1;

        while (num > 0) {
            int digit = num % 10;
            sum += digit;
            product *= digit;
            num /= 10;
        }

        return sum < product;
    }
}